staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -742,16 +747,35 @@ func validateCustomResourceDefinitionValidation(ctx context.Context, customResou
 	return allErrs
 }
 
+type costInfo struct {
+	// Cardinality tracks the number of times a rule in the current schema node will actually get executed
+	// due to being the child of arrays/maps/etc.
+	Cardinality *uint64
+}
+
 var metaFields = sets.NewString("metadata", "kind", "apiVersion")
 
 // ValidateCustomResourceDefinitionOpenAPISchema statically validates
-func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSchemaProps, fldPath *field.Path, ssv specStandardValidator, isRoot bool, opts *validationOptions) field.ErrorList {
+func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSchemaProps, fldPath *field.Path, ssv specStandardValidator, isRoot bool, opts *validationOptions, parentNode *costInfo) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	if schema == nil {
 		return allErrs
 	}
 
+	cardinality := extractMaxElements(schema)
+	if parentNode != nil && cardinality != nil {
+		if parentNode.Cardinality != nil {
+			*cardinality *= *parentNode.Cardinality
+		} else {
+			cardinality = nil
```

Comment:
Why does a passed in costInfo that is non-nil but contains a cardinality that is nil, result in a nil cardinality being passed down to children nodes?